battenberg_gitlab/apply.py

```diff
@@ -0,0 +1,101 @@
+import os
+import logging
+import json
+from typing import List
+from functools import reduce
+from gitlab import Gitlab
+from pygit2 import Repository
+from battenberg import Battenberg
+from battenberg.errors import MergeConflictException
+from battenberg_gitlab.search import get_projects
+from battenberg_gitlab.utils import clone_or_discover_repo, ensure_workspace
+
+
+logger = logging.getLogger(__name__)
+
+
+# TODO Remove these once we've we've backfilled .cookiecutter.json
+with open(os.path.join(os.path.dirname(__file__), '..', 'templates.json')) as f:
```

Comment:
This templates.json seems someone magical in that it contains all the initial values used to cookie cut an archetype project?

Reply:
No it's just a temporary file that contains the contents of the current projects template context as only in the round of PRs for v1.1.0 I actually added the `.cookiecutter.json` files for each project so this was me easily adding them :) 

Also see: https://github.com/zillow/battenberg-gitlab/pull/1#discussion_r339224825